Source/ui_unix/ControllerConfig/inputeventselectiondialog.cpp

```diff
@@ -103,9 +103,82 @@ void InputEventSelectionDialog::SetupInputDeviceManager(std::unique_ptr<CGamePad
 		}
 	};
 
-	GPDL.get()->UpdateOnInputEventCallback(onInput);
+	GPDL->UpdateOnInputEventCallback(onInput);
 }
 
+#elif defined(__APPLE__)
+void InputEventSelectionDialog::SetupInputDeviceManager(CGamePadDeviceListener* GPDL)
+{
+	auto onInput = [=](std::array<unsigned int, 6> device, int code, int value, int type) -> void {
+		bool is_axis = type > 1;
+		if(!is_axis)
+		{
+			if(!setCounter(value)) return;
+		}
+		QString key = QString("btn-").append(QString::number(code));
+		if(is_axis)
+		{
+			if(type != 2)
+			{
+				CFIndex triggerRange = (255 * 20) / 100;
+				CFIndex triggerVal1 = 255 - triggerRange;
+				CFIndex triggerVal2 = 0 + triggerRange;
+				if(value < triggerVal1 && triggerVal2 < value)
+				{
+					setCounter(0);
+					return;
+				}
+				setCounter(1);
+				setSelectedButtonLabelText("Selected Key: " + key);
+				m_key1.id = code;
+				m_key1.device = device;
+				m_key1.type = type;
+				m_key1.bindtype = CInputBindingManager::BINDINGTYPE::BINDING_SIMPLE;
+			}
+			else
+			{
+				m_key1.id = code;
+				m_key1.device = device;
+				m_key1.type = type;
+				m_key1.value = value;
```

Comment:
Don't know if that applies to DS3/DS4, but it seems I need a check for value being less than 8 here. My Moga Pro 2 reports value = 15 when the dpad is being released. Values are in range [0, 8[ when pressing the directions.

Reply:
DS3 DPad is reported as individual digital buttons (to be more exact reported as a bit value per direction).
DS4 reports dpad as a range from 0-7 and 8 when released/neutral.
so the DS4 would have similar issue, however, since value is only reported when changed, I didn't find it to be an issue(aka it's only an issue when configuring, but clicking any other button and holding or even releasing would cause it to be ignored, granted not my finest solutions).

Reply:
Well, adding a check for `value < 8` would probably be a good solution. If it's less than 8, you call `setCounter(1)` and call `setCounter(0)` otherwise. That's what I did on my end and it seems to do the job.